src/file_pool.cpp

```diff
@@ -239,16 +239,18 @@ namespace libtorrent
 	{
 		std::unique_lock<std::mutex> l(m_mutex);
 
-		auto begin = m_files.lower_bound(std::make_pair(st, file_index_t(0)));
+		auto const begin = m_files.lower_bound(std::make_pair(st, file_index_t(0)));
+		auto current = begin;
```

Comment:
if you're introducing a loop variable, I think the loop below should be turned into a for-loop. (and ``current`` could be named ``i``)